### DISCUSSION TITLE
Best practice for keeping package recipe up to date in each new release of spack

### DISCUSSION THREAD

**Discussion by markcmiller86** [spack/spack#30206] [Q&A]
As we make releases of our package, we'd like to keep spack's recipe for it up to date.

What is the proper process for doing that? I mean, do we just submit PRs for the respective `package.py` file to `develop` after we've released our package?

Or, should we do that on an upcoming release candidate? Or both?

If they are done only on `develop` about how long would the new `package.py` be there before the next public release of spack from `develop` is made?

Is there a place we should go to track anticipated release schedules and branches of development?

**Comment by cessenat** [comment]
From my experience and from https://spack.readthedocs.io/en/latest/developer_guide.html#release-branches
you only need to submit a PR to develop ; if your changes alter concretization they will not be included in the next spack minor release but they will to the next major. Next major and beyond are spack projects https://github.com/spack/spack/projects which are ready when status bar is completely green. About timing and anticipation I cannot tell. Personally I use spack updates of the develop branch with very rare trouble (usually typically with openssl that systematically deprecates previous releases).
